manage/import-export-dropbox.js

```diff
@@ -0,0 +1,191 @@
+/* global messageBox */
+'use strict';
+
+const DROPBOX_API_KEY = 'uyfixgzre8v1bkg';
+const FILENAME_ZIP_FILE = 'stylus.json';
+const DROPBOX_FILE = 'stylus.zip';
+const API_ERROR_STATUS_FILE_NOT_FOUND = 409;
+const HTTP_STATUS_CANCEL = 499;
+
+function messageProgressBar(data) {
+  return messageBox({
+    title: `${data.title}`,
+    className: 'config-dialog',
+    contents: [
+      $create('p', data.text)
+    ],
+    buttons: [{
+      textContent: t('confirmClose'),
+      dataset: {cmd: 'close'},
+    }],
+  }).then(_ => {
+    document.body.style.minWidth = '';
+    document.body.style.minHeight = '';
+  });
+}
+
+function hasDropboxAccessToken() {
+  return chromeLocal.getValue('dropbox_access_token');
+}
+
+function requestDropboxAccessToken() {
+  const browserApi = typeof browser === 'undefined' ? chrome : browser;
+  const client = new Dropbox.Dropbox({clientId: DROPBOX_API_KEY});
+  const authUrl = client.getAuthenticationUrl(browserApi.identity.getRedirectURL());
+
+  return browserApi.identity.launchWebAuthFlow({url: authUrl, interactive: true})
+  .then(urlReturned => {
+    const params = new URLSearchParams(new URL(urlReturned).hash.replace('#', '?'));
+
+    chromeLocal.setValue('dropbox_access_token', params.get('access_token'));
+
+    return params.get('access_token');
+  });
+}
+
+function uploadFileDropbox(client, stylesText) {
+  return client.filesUpload({path: '/' + DROPBOX_FILE, contents: stylesText});
+}
+
+$('#sync-dropbox-export').onclick = () => {
+
+  messageProgressBar({title: t('bckpDropboxStyles'), text: t('connectingDropbox')});
+
+  hasDropboxAccessToken().then(token => {
+    if (typeof token === 'undefined') {
+      return requestDropboxAccessToken();
+    }
+
+    return token;
+  })
+  .then(token => {
+    const client = new Dropbox.Dropbox({
+      clientId: DROPBOX_API_KEY,
+      accessToken: token
+    });
+
+    return client.filesDownload({path: '/' + DROPBOX_FILE})
+     .then(_ => messageBox.confirm(t('overwriteFileExport')))
+    .then(ok => {
+      /** deletes file if user want to */
+      if (!ok) {
+        return Promise.reject({status: HTTP_STATUS_CANCEL});
+      }
+
+      return client.filesDelete({path: '/' + DROPBOX_FILE});
+    })
+    /** file deleted with success, get styles and create a file */
+    .then(_ => {
+      messageProgressBar({title: t('bckpDropboxStyles'), text: t('gettingStyles') });
```

Comment:
* no need to declare an unused parameter, please use `() => `
* extract t('bckpDropboxStyles') as `title` const